core/src/main/java/org/mskcc/oncotree/crosswalk/MSKConceptCache.java

```diff
@@ -67,20 +70,38 @@ public MSKConcept get(String oncoTreeCode) {
     private void resetCache() {
         logger.info("resetCache() -- clearing Crosswalk MSKConcept cache and refilling");
         oncoTreeCodesToMSKConcepts.clear();
-        List<OncoTreeNode> oncoTreeNodes = oncoTreeRepository.getOncoTree(VersionUtil.getDefaultVersion());
-        for (OncoTreeNode node : oncoTreeNodes) {
-            getFromCrosswalkAndSave(node.getCode());
-        } 
+        // versions are ordered in ascending order by release date
+        for (Version version : VersionUtil.getVersions()) {
+            List<OncoTreeNode> oncoTreeNodes = oncoTreeRepository.getOncoTree(version);
+            for (OncoTreeNode node : oncoTreeNodes) {
+                MSKConcept mskConcept = getFromCrosswalkAndSave(node.getCode());
+                // get all codes defined so far for this topbraid uri and save in history
+                if (topBraidURIsToOncotreeCodes.containsKey(node.getURI())) {
+                    // do not add this code to the history, but add any others
+                    HashSet<String> allButThisNode = new HashSet<String>(topBraidURIsToOncotreeCodes.get(node.getURI()));
```

Comment:
because we are going in forward time order, this "snapshot" will show all the oncotree codes defined up to this point in time. (will only include current and previous codes)